sim/global-types.ts

```diff
@@ -23,6 +23,7 @@ type ID = '' | string & {__isID: true};
 type PokemonSlot = '' | string & {__isSlot: true};
 interface AnyObject {[k: string]: any}
 
+type AllStatIDs = 'currenthp' | StatID;
```

Comment:
I consider `AllStatIDs` a pretty bad name for this, tbh. Probably just use `StatID | 'currenthp'`, you only really need it in like two places. Honestly probably use `StatIDExceptHP | 'maxhp' | 'currenthp'` for extra clarity.

Reply:
ah, Good point. I was using it in a lot more places before I swapped things to using the array. Good catch

Reply:
How about StatIDAndCurrentHP?

Reply:
In this case, since it's not generic, I would go with `DamageCalcStatID`.

But I think given the other context, I would rather remove `maxhp` and `currenthp` from the type entirely.